src/test/java/bio/terra/service/auth/iam/sam/SamIamTest.java

```diff
@@ -759,6 +759,105 @@ void testPatchAuthDomain() throws Exception {
           .patchAuthDomainV2(
               IamResourceType.DATASNAPSHOT.getSamResourceName(), snapshotId.toString(), authDomain);
     }
+
+    @Test
+    void setResourceParent() throws ApiException, InterruptedException {
+      UUID childId = UUID.randomUUID();
+      UUID parentId = UUID.randomUUID();
+      samIam.setResourceParent(
+          TEST_USER.getToken(),
+          IamResourceType.DATASNAPSHOT,
+          childId,
+          IamResourceType.DATASET,
+          parentId);
+      verify(samResourceApi)
+          .setResourceParent(
+              IamResourceType.DATASNAPSHOT.getSamResourceName(),
+              childId.toString(),
+              new FullyQualifiedResourceId()
+                  .resourceTypeName(IamResourceType.DATASET.getSamResourceName())
+                  .resourceId(parentId.toString()));
+    }
+
+    @Test
+    void setResourceParentThrows() throws ApiException {
+      UUID childId = UUID.randomUUID();
+      UUID parentId = UUID.randomUUID();
+      ApiException samEx =
+          new ApiException(HttpStatusCodes.STATUS_CODE_NOT_FOUND, "Resource not found");
+      doThrow(samEx)
+          .when(samResourceApi)
+          .setResourceParent(
+              IamResourceType.DATASNAPSHOT.getSamResourceName(),
+              childId.toString(),
+              new FullyQualifiedResourceId()
+                  .resourceTypeName(IamResourceType.DATASET.getSamResourceName())
+                  .resourceId(parentId.toString()));
+      assertThrows(
+          IamNotFoundException.class,
+          () ->
+              samIam.setResourceParent(
+                  TEST_USER.getToken(),
```

Comment:
I'm sure you'll get a sonar warning about this, but the goal is that there's only one function call in the lambda passed to `assertThrows()`, so we know exactly what function is generating the exception. The fix is to assign `TEST_USER.getToken()` to a local variable outside the lambda.